src/ui/components/dialog/DialogProvider.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { createContext, useState, useRef, useContext, useCallback, PropsWithChildren, ReactNode, cloneElement, ReactElement } from 'react';
+import { Dialog } from '@ui/components/dialog';
+
+export type DialogOption = {
+  fullWidth?: boolean;
+  enableBackdropClick?: boolean;
+};
+
+type DialogSubmitResult = any;
```

Comment:
P3: unknown타입으로 명시하는게 any보다 명시적이기에 이렇게 변경해도 괜찮을 것 같아서 추천드려요

```suggestion
type DialogSubmitResult = unknown;
```

Reply:
> P3: unknown타입으로 명시하는게 any보다 명시적이기에 이렇게 변경해도 괜찮을 것 같아서 추천드려요

unknown 타입을 이용해서 해보겠습니다!